src/muya/lib/parser/marked/lexer.js

```diff
@@ -302,7 +314,9 @@ Lexer.prototype.token = function (src, top) {
         // Determine whether item is loose or not. If previous item is loose
         // this item is also loose.
         loose = next = next || /^ *([*+-]|\d{1,9}(?:\.|\)))( +\S+\n\n(?!\s*$)|\n\n(?!\s*$))/.test(itemWithBullet)
-
+        // A list is loose if any of its constituent list items are separated by blank lines,
+        // or if any of its constituent list items directly contain two block-level elements with a blank line between them. 
+        loose = next = next || /\n\n(?!\s*$)/.test(item)
```

Comment:
This is a duplicate check because the regex above checks the same in the last `OR` statement.

Reply:
I think this is a little different from above regex, because this fix example https://spec.commonmark.org/0.29/#example-224.

>/\n\n(?!\s*$)/.test(item)

will test if a blank line is between item.

and above regexp can not test this?

Reply:
How about only use the second regexp `/\n\n(?!\s*$)/`? I just run test with only the second regexp, it still failed 126 commonmark examples.